browser/ui/tabs/split_view_browser_data.cc

```diff
@@ -107,4 +108,72 @@ void SplitViewBrowserData::RemoveObserver(
   observers_.RemoveObserver(observer);
 }
 
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure() = default;
+
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure(
+    base::WeakPtr<SplitViewBrowserData> data,
+    base::OnceClosure closure)
+    : data_(data),
+      closure_(
+          std::make_unique<base::ScopedClosureRunner>(std::move(closure))) {
+  CHECK(data_);
+}
+
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure(
+    SplitViewBrowserData::OnTabDragEndedClosure&& other) noexcept {
+  RunCurrentClosureIfNeededAndReplaceWith(std::move(other));
+}
+
+SplitViewBrowserData::OnTabDragEndedClosure&
+SplitViewBrowserData::OnTabDragEndedClosure::operator=(
+    SplitViewBrowserData::OnTabDragEndedClosure&& other) noexcept {
+  RunCurrentClosureIfNeededAndReplaceWith(std::move(other));
+  return *this;
+}
+
+void SplitViewBrowserData::OnTabDragEndedClosure::RunAndReset() {
+  if (closure_) {
+    this->closure_->RunAndReset();
+  }
+  data_.reset();
+}
+
+void SplitViewBrowserData::OnTabDragEndedClosure::
+    RunCurrentClosureIfNeededAndReplaceWith(OnTabDragEndedClosure&& other) {
+  if (this->data_.get() == other.data_.get()) {
+    // In case |this| and |other| are pointing at the same |data_|, just discard
+    // the old one. This means we've got the callback from the same Browser
+    // instance.
+    if (this->closure_) {
+      this->closure_->Release().Reset();
```

Comment:
Maybe `Reset()` is redundant?
```suggestion
      this->closure_->Release();
```

Reply:
I believe Reset() is needed due to [[nodiscard]] attribute on the closure.